static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -102,11 +98,13 @@ const getTestimonialBlock = (
 const defaultPageCheck = (page: Object): boolean =>
     page.contentType === 'Article' && !page.isMinuteArticle;
 
-const defaultCanEpicBeDisplayed = (test: ContributionsABTest): boolean => {
-    const canReasonablyAskForMoney =
-        test.showToContributorsAndSupporters ||
-        commercialFeatures.canReasonablyAskForMoney;
+const shouldShowReaderRevenue = (
+    showToContributorsAndSupporters: ?boolean = false
+): boolean =>
+    (userShouldSeeReaderRevenue || showToContributorsAndSupporters) &&
```

Comment:
yes you're right! I think it's some other subtlety of flow that means this type-checks as is. once I actually call `userShouldSeeReaderRevenue()`, then suddenly flow complains that `showToContributorsAndSupporters` could be null or undefined - so you're right to question the maybe type above (I thought I should put a maybe type on a parameter with a default value to express the fact that the caller doesn't have to supply it - now I realise it actually means that null or undefined are valid values, which isn't the case here).

so for some reason, using the `||` operator with a function reference and a maybe boolean type-checks, but `||` with a boolean and a maybe boolean doesn't. weird